pygmt/tests/test_grdfill.py

```diff
@@ -114,6 +114,31 @@ def test_grdfill_gridfill_dataarray(grid):
     npt.assert_array_equal(result[3:6, 3:5], bggrid[3:6, 3:5])
 
 
+def test_grdfill_hole(grid, expected_grid):
+    """
+    Test grdfill with different hole values.
+    """
+    # Prepare for a grid with -99999 as holes.
+    grid_no_nan = grdfill(grid=grid, constantfill=-99999)
+    assert not np.isnan(grid_no_nan).any()
+    assert -99999 in grid_no_nan
+    # Now that we have a grid with -99999 as holes, fill it with a constant value of 20.
+    result = grdfill(grid=grid_no_nan, constantfill=20, hole=-99999)
+
+    # check information of the output grid
+    assert isinstance(result, xr.DataArray)
+    assert result.gmt.gtype is GridType.GEOGRAPHIC
+    assert result.gmt.registration is GridRegistration.PIXEL
+    # check information of the output grid
+    xr.testing.assert_allclose(a=result, b=expected_grid)
```

Comment:
Maybe it is enough to have this comment once.

```suggestion
    # Check information of the output grid
    assert isinstance(result, xr.DataArray)
    assert result.gmt.gtype is GridType.GEOGRAPHIC
    assert result.gmt.registration is GridRegistration.PIXEL
    xr.testing.assert_allclose(a=result, b=expected_grid)
```